torch_frame/nn/encoder/stype_encoder.py

```diff
@@ -208,7 +207,20 @@ def na_forward(self, feat: TensorData) -> TensorData:
                     feat.device)
             else:
                 raise ValueError(f"Unsupported NA strategy {self.na_strategy}")
-            column_data[nan_mask] = fill_value
+            if isinstance(feat, _MultiTensor):
+                feat.fillna_col(col, fill_value)
+            else:
+                column_data[nan_mask] = fill_value
+        # Add better safeguard here to make sure nans are actually
+        # replaced, expecially when nans are represented as -1's. They are
+        # very hard to catch as they won't error out.
+        filled_values = feat
+        if isinstance(feat, _MultiTensor):
+            filled_values = feat.values
+        if filled_values.is_floating_point():
+            assert not torch.isnan(filled_values).any()
+        else:
+            assert not (filled_values == -1).any()
```

Comment:
Thanks for being extra careful! just curious, do you know when this can happen after this fix? or is it mainly for safeguarding?

Reply:
Also, I am wondering if this code can handle `dict[str, MultiNestedTensor]` (currently, we don't allow NaN since it's a tokenized results).

Reply:
It's mainly for safeguarding. Also I don't know if `dict[str, MultiNestedTensor]` will hit this code because we don't allow specifying `NAStrategy` for `text_tokenized`.

Reply:
> Also I don't know if dict[str, MultiNestedTensor] will hit this code because we don't allow specifying NAStrategy for text_tokenized.
Ok makes sense. Then no worries for now.